sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/models/AmqpMessageId.java

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.models;
+
+import java.util.Objects;
+
+/**
+ * This represents Message id. Amqp specification support message id in various types. This class only implements
+ * {@link String} representation at present.
+ *
+ * <p><strong>Create and retrieve message id </strong></p>
+ * {@codesnippet com.azure.core.amqp.models.AmqpBodyType.checkBodyType}
+ *
+ * <b>Amqp message id types:</b>
+ * <ul>
+ * <li><a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-message-id-string">String</a></li>
+ * <li><a href="http://docs.oasis-open.org/amqp/core/v1.0/amqp-core-messaging-v1.0.html#type-message-id-ulong">long</a></li>
+ * <li><a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-message-id-uuid">Uuid</a></li>
+ * <li><a href="http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-message-id-binary">binary</a></li>
```

Comment:
If the message ids support string, long, uuid and binary types why does the constructor only have `String`?

Reply:
We ,including dotnet, are only implementing `String` in this GA release.   And other types will be supported in future and by all the languages as well.